### PR TITLE
Removed internal channels and replaced with a proper queue

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

# Changes

- :broom: Remove internal channels and replaced with an unbounded blocking queue

/kind <kind>

Fixes #

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

File: pkg/network/message_queue.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
+package network
+
+import (
+	"context"
+	"sync"
+
+	ctrl "knative.dev/control-protocol/pkg"
+)
+
+type unboundedMessageQueue struct {
```
Comment (matzew) [inline pkg/network/message_queue.go:10]: no built-in lib for this?

**Comment by slinkydeveloper** [comment]
/hold

I'm adding a test to increase the coverage

File: pkg/network/message_queue.go (hunk 1 above)
Comment (slinkydeveloper) [reply pkg/network/message_queue.go:11]: Added a comment about that

**Comment by slinkydeveloper** [comment]
/unhold

Added a client redial test, now we have some proper coverage

**Comment by slinkydeveloper** [comment]
/hold

I want to make the queue simpler

**Comment by slinkydeveloper** [comment]
/unhold

The change is too big, I'm gonna do that in another PR

File: pkg/network/message_queue.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import (
+	"context"
+	"sync"
+
+	ctrl "knative.dev/control-protocol/pkg"
+)
+
+// unboundedMessageQueue is a pull based message queue.
+// If you're wondering why we need this,
+// we're waiting for https://github.com/golang/go/issues/27935 to happen.
```
Comment (matzew) [inline pkg/network/message_queue.go:12]: :+1:

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Pull request merged**